### ISSUE TITLE
Entity spawning inside Tardis

### ISSUE THREAD

**Issue by Mysterypancake1** [MattJeanes/TARDIS#276]
Currently, if you are looking through the Tardis doors and you try to spawn an entity, it doesn't spawn it inside the Tardis. It spawns on the Tardis entity. This is perfectly understandable, but it would add realism if you could detour this function to change the position to the Tardis interior. You appear to already have detoured the bullet functions so if you shoot into the Tardis the bullets end up there, so it may be possible to do this as well.

**Comment by MattJeanes** [comment]
Yeah I agree with that, not sure if it's possible to detour spawn positions but we'll see. Tracing will need to improve first though, it's currently not very good, but its more difficult to tell with the bullets.

**Comment by EdusgprNetwork** [comment]
Something _sorta_ similar happens when you use the rocket launcher from the interior. The rocket detours from where you're looking and goes to the interior instead

**Comment by Noname12345GER** [comment]
local tr = GetEyeTrace(ply)
If tr.Entity == IsDoors() or IsExterior() then
hook.Call(TARDISSpawnOverride)
end

function (OnSpawn, TardisSpawnOverride)
ENT:SetPos(Doors.GetViewPos())
end

I have seen you can get the position in the world if the player looks thru the doors. The hook might be wrong, i am not working with Hooks that often. This is just a quick idea.
